jupyterhub/schema.yaml

```
@@ -1160,6 +1160,43 @@ properties:
             patternProperties: *labels-and-annotations-patternProperties
             description: |
               Kubernetes annotations to apply to the k8s ServiceAccount.
+      extraPodSpec: &jupyterhub-extra-pod-spec
+        type: object
+        additionalProperties: true
+        description: |
+          Arbitrary extra k8s pod specification as a YAML object. The default
+          value of this setting is an empty object, i.e. no extra configuration.
+          The value of this property is merged into the pod specification as-is.
+
+          This is a powerful tool for expert k8s administrators with advanced
+          configuration requirements. This setting should only be used for
+          configuration that cannot be accomplished through the other settings.
+          Misusing this setting can break your deployment and/or compromise
+          your system security.
+
+          One use of this setting is to configure the hub pod with host networking:
+
+          ```yaml
+          hub:
+            extraPodSpec:
+              hostNetwork: true
+              dnsPolicy: ClusterFirstWithHostNet
+          ```
+
+          To configure the proxy pod with host networking:
+
+          ```yaml
+          proxy:
+            chp:
+              extraPodSpec:
+                hostNetwork: true
+                dnsPolicy: ClusterFirstWithHostNet
+          ```
+
+          and so on...
```

Since this will be showing up in the documentation four times, one for the hub, proxy.chp, proxy.traefik, and scheduling.userScheduler, I figure we can reduce the examples provided to only one.

```suggestion
          One use of this setting is to configure the hub pod with host networking:

          ```yaml
          hub:
            extraPodSpec:
              hostNetwork: true
              dnsPolicy: ClusterFirstWithHostNet
          ```
```

---

I like the idea of having a concrete example in the documentation. However, if you look at the documentation for `proxy.chp.extraPodSpec`, it gives an example for "this setting" that is not for "this setting". See what I mean? I think it's a bit awkward. I'm going to think about this one a bit more.

---

Yes it is a bit awkward to not adjust the text based on setting =\ i dont know of a way that feels viable to fix that.

We can retain this example as it is also btw, not strongly opinionated about this. Let me know if you think the state is ok to merge from your end.